polytracker/taint_dag.py

```diff
@@ -117,17 +118,15 @@ def __init__(self, buffer: bytearray) -> None:
 
     def read_fd_headers(self) -> Iterator[Tuple[str, TDFDHeader]]:
         assert self.header.fd_mapping_offset > 0
-        assert self.header.fd_mapping_size > 0
+        assert self.header.fd_mapping_count > 0
 
         offset = self.header.fd_mapping_offset
-        end = offset + self.header.fd_mapping_size
-
-        while offset < end:
-            fdhdr = TDFDHeader.from_buffer_copy(self.buffer, offset)
-            offset += sizeof(TDFDHeader)
-            path = str(self.buffer[offset : offset + fdhdr.namelen], "utf-8")
+        for i in range(0, self.header.fd_mapping_count):
+            header_offset = offset + sizeof(TDFDHeader)*i
+            fdhdr = TDFDHeader.from_buffer_copy(self.buffer, header_offset)
+            sbegin = offset + fdhdr.name_offset
```

Comment:
Am I correct in saying that the file name offset is relative to the start of the fd header section? Given the computation above.

Reply:
Yes. See also the comment in fdmapping.hpp.